src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,8 +102,14 @@ public VSSolutionManager()
                 ErrorHandler.ThrowOnFailure(hr);
             }
 
-            UserAgent.SetUserAgentString(
-                new UserAgentStringBuilder().WithVisualStudioSKU(VSVersionHelper.GetFullVsVersionString()));
+            // Allow this constructor to be invoked from either the UI or a worker thread
+            ThreadHelper.JoinableTaskFactory.Run(async delegate
```

Comment:
So in order to support Xamarin we're about to sacrifice perf in the rest 99% of cases here?